common/interpreter/formatSwitch.go

```diff
@@ -61,6 +85,58 @@ func (s FormatSwitch) ParseJSON(data []byte) ErrorDescriptor { // nolint:ireturn
 	}
 }
 
+// We are dealing with the list of error objects.
+// Each error object will be mapped to the template.
+//
+// Ex: provider response is array:
+// [
+//
+//	{"code":128, "message":"conflicting entries"},
+//	{"code":70, "message":"invalid time format", "description":"use ISO nano"},
+//
+// ].
+func (s FormatSwitch) mapObjectsToErrorList(payloads []map[string]any) ErrorDescriptor {
+	result := &listErrorDescriptor{
```

Comment:
rename to `error` or `errorDescriptior` or `errDesc` or something that indicates this is error.

Reply:
It is not exactly an error. It is a struct that knows how to wrap, decorate an error.
```
type ErrorDescriptor interface {
	CombineErr(base error) error
}
```
listErrorDescriptor will be a combined list of errors with common "base", reason.